src/main/java/dev/slimevr/vr/processor/skeleton/LegTweaks.java

```diff
@@ -0,0 +1,822 @@
+		this.enabled = enabled;
+	}
+
+	public void setFloorclipEnabled(boolean floorclipEnabled) {
+		this.floorclipEnabled = floorclipEnabled;
+		// reset the buffer
+		this.bufferHead = new LegTweakBuffer();
+	}
+
+	public void setSkatingReductionEnabled(boolean skatingCorrectionEnabled) {
+		this.skatingCorrectionEnabled = skatingCorrectionEnabled;
+		// reset the buffer
+		this.bufferHead = new LegTweakBuffer();
+		this.bufferInvalid = true;
+	}
+
+	public boolean getEnabled() {
+		return this.enabled;
+	}
+
+	public boolean getFloorclipEnabled() {
+		return this.floorclipEnabled;
+	}
+
+	public boolean getSkatingReductionEnabled() {
+		return this.skatingCorrectionEnabled;
+	}
+
+	public void resetBuffer() {
+		bufferInvalid = true;
+	}
+
+	// set the vectors in this object to the vectors in the skeleton
+	private void setVectors() {
+		// set the positions of the feet and knees to the skeletons current
+		// positions
+		if (skeleton.computedLeftKneeTracker != null || skeleton.computedRightKneeTracker != null) {
+			kneesActive = true;
+			leftKneePosition = skeleton.computedLeftKneeTracker.position;
+			rightKneePosition = skeleton.computedRightKneeTracker.position;
+		} else {
+			kneesActive = false;
+			leftKneePosition = leftKneePlaceholder;
+			rightKneePosition = rightKneePlaceholder;
+		}
+		if (
+			skeleton.computedLeftFootTracker != null
+				&& skeleton.computedRightFootTracker != null
+				&& skeleton.computedWaistTracker != null
+		) {
+			leftFootPosition = skeleton.computedLeftFootTracker.position;
+			rightFootPosition = skeleton.computedRightFootTracker.position;
+			waistPosition = skeleton.computedWaistTracker.position;
+			leftFootRotation = skeleton.computedLeftFootTracker.rotation;
+			rightFootRotation = skeleton.computedRightFootTracker.rotation;
+		}
+
+		// get the vector for acceleration of the feet and knees
+		// (if feet are not available, fallback to 6 tracker mode)
+		if (skeleton.leftFootTracker != null && skeleton.rightFootTracker != null) {
+			skeleton.leftFootTracker.getAcceleration(leftFootAcceleration);
+			skeleton.rightFootTracker.getAcceleration(rightFootAcceleration);
+		} else {
+			leftFootAcceleration.set(0, 0, 0);
+			rightFootAcceleration.set(0, 0, 0);
+		}
+		if (skeleton.leftLowerLegTracker != null && skeleton.rightLowerLegTracker != null) {
+			skeleton.leftLowerLegTracker.getAcceleration(leftLowerLegAcceleration);
+			skeleton.rightLowerLegTracker.getAcceleration(rightLowerLegAcceleration);
+		} else {
+			leftLowerLegAcceleration.set(0, 0, 0);
+			rightLowerLegAcceleration.set(0, 0, 0);
+		}
+
+	}
+
+	// updates the object with the latest data from the skeleton
+	private boolean preUpdate() {
+		// populate the vectors with the latest data
+		setVectors();
+
+		// if not initialized, we need to initialize floor level and waist to
+		// floor distance (must happen immediately after reset)
+		if (!initialized) {
+			floorLevel = (leftFootPosition.y + rightFootPosition.y) / 2f + 0.02f;
```

Comment:
Can there be an explanation for what and why `0.02f` is here? It looks like a magic number to me.

Reply:
Yes, it has been demystified